### PR TITLE
Planck Keymap: Inkwell

### PR DESCRIPTION
**Pull request by kevinhartley**
## Description

Plank layout addition with Base, Lower, Raise, Adjust, and 1 encoder. 

- Features:
- Numberpad
- Symbol Pad
- Programming utilities: ()[]{}<>
- Mouse Control
- Function Pad
- Encoder:
> - Volume (default), 
> - Brightness (shift+turn), 
> - Scroll Y(RSE+turn), 
> - Scroll X (RSE+shift+turn), 
> - Zoom (push+turn)
> - Play/Pause (double click)

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* none

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/planck/keymaps/inkwell/config.h [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#pragma once
```
Comment (drashna) [inline keyboards/planck/keymaps/inkwell/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/planck/keymaps/inkwell/keymap.c [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+enum planck_layers {
+  _QWERTY,
+  _LOWER,
+  _RAISE,
+  _ADJUST
+};
+
+enum planck_keycodes {
+  QWERTY = SAFE_RANGE
+};
```
Comment (drashna) [inline keyboards/planck/keymaps/inkwell/keymap.c:30]: If you only have the single default layer, this isn't needed, at all. 
```suggestion
```

File: keyboards/planck/keymaps/inkwell/keymap.c [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    case QWERTY:
+      if (record->event.pressed) {
+        print("mode is qwerty\n");
+        set_single_persistent_default_layer(_QWERTY);
+      }
+      return false;
+      break;
+  }
+  return true;
+}
```
Comment (drashna) [inline keyboards/planck/keymaps/inkwell/keymap.c:133]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/planck/keymaps/inkwell/keymap.c (hunk 2 above)
Comment (kevinhartley) [reply keyboards/planck/keymaps/inkwell/keymap.c:30]: Sounds great, thank you!

File: keyboards/planck/keymaps/inkwell/keymap.c (hunk 3 above)
Comment (kevinhartley) [reply keyboards/planck/keymaps/inkwell/keymap.c:133]: 👍

File: keyboards/planck/keymaps/inkwell/config.h (hunk 1 above)
Comment (kevinhartley) [reply keyboards/planck/keymaps/inkwell/config.h:17]: Done!

**Review by drashna — APPROVED** [review]

File: keyboards/planck/keymaps/inkwell/config.h [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+   - MIDI notes can be added to the keymap
+   - Octave shift and transpose
+   - Virtual sustain, portamento, and modulation wheel
+   - etc.
+*/
+//#define MIDI_ADVANCED
+
+/* override number of MIDI tone keycodes (each octave adds 12 keycodes and allocates 12 bytes) */
+//#define MIDI_TONE_KEYCODE_OCTAVES 2
+
```
[the remark this answers is not in the document]
Comment (kevinhartley) [reply keyboards/planck/keymaps/inkwell/config.h]: Done!

**Review by fauxpark — APPROVED** [review]

**Comment by kevinhartley** [comment]
@drashna are you aware of who is authorized to merge? Thanks!

**Comment by drashna** [comment]
Thanks!

**Pull request merged**